hack/create_graph_schema.py

```diff
@@ -0,0 +1,46 @@
+import os
+import requests
+import json
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+GREMLIN_SERVER_URL_REST = "http://{host}:{port}".format(
+                          host=os.environ.get("BAYESIAN_GREMLIN_HTTP_SERVICE_HOST", "localhost"),
+                          port=os.environ.get("BAYESIAN_GREMLIN_HTTP_SERVICE_PORT", "8182"))
+
+GRAPH_INITIALIZED = False
+SCHEMA_CREATED = False
+# wait for graph to be initialized
+while not GRAPH_INITIALIZED:
+    try:
+        print("Calling GREMLIN - " + GREMLIN_SERVER_URL_REST)
+        resp = requests.get(GREMLIN_SERVER_URL_REST)
+        if resp.status_code >= 200:
+            logger.info("Gremlin Instance available")
+            GRAPH_INITIALIZED = True
+            break
+    except:
+        print("Waiting for Gremlin HTTP to be initialized. Sleeping for 10 seconds")
+        time.sleep(10)
+        continue
+
+# send the schema payload for creation
+current_file_path = os.path.dirname(os.path.realpath(__file__))
+schema_file_path = os.path.join(current_file_path, 'schema.groovy')
+with open(schema_file_path, 'r') as f:
+    str_gremlin_dsl = f.read()
+
+while not SCHEMA_CREATED:
+    print("Creating Graph Schema Now...")
+    payload = {'gremlin': str_gremlin_dsl}
+
+    response = requests.post(GREMLIN_SERVER_URL_REST, json=payload)
+    json_response = response.json()
+
+    if response.status_code != 200:
```

Comment:
I wonder what happened here if gremlin returns `201`. It fails with `RuntimeError`, but will it be consistent when we want to create same scheme again even though it was successful last time?

Reply:
Yes. If schema is not created, it will create afresh.  Even if schema is created in the first place, it will just pass through but still send 200. Plus, if its not 200, then gremlin rolls back the entire transaction.

Reply:
sounds good to me. This sounds like a good test case to me.